### PR TITLE
elixir: add Elixir Enum best practice

### PR DESCRIPTION
**Pull request by wingyplus**
Add 2 rules for Enum:

1. detect and rewrite `Enum.map/2 |> Enum.into/2` into `Enum.into/3`.
2. detect and rewrite `Enum.map/2 |> Enum.join/2` into `Enum.map_join/3`.

### CODE REVIEW

File: elixir/lang/best-practice/enum.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      $E
+      |> Enum.into($INTO, $FUN end)
+    metadata:
+      references:
+        - https://github.com/rrrene/credo/blob/master/lib/credo/check/refactor/map_into.ex
+      category: best-practice
+      technology:
+        - elixir
+
+  - id: enum_map_join
```
Comment (colleend) [inline elixir/lang/best-practice/enum.yaml]: Separate the two rules into 2 files, and separate the related test files too 👍

File: elixir/lang/best-practice/enum.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+rules:
+  - id: enum_map_into
+    message: >-
+      Use `Enum.into/3` is more efficient than using `Enum.map/2 |> Enum.into/2`.
```
Comment (colleend) [inline elixir/lang/best-practice/enum.yaml]:
```suggestion
      Using `Enum.into/3` is more efficient than using `Enum.map/2 |> Enum.into/2`. 
```

File: elixir/lang/best-practice/enum.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    metadata:
+      references:
+        - https://github.com/rrrene/credo/blob/master/lib/credo/check/refactor/map_into.ex
+      category: best-practice
+      technology:
+        - elixir
+
+  - id: enum_map_join
+    message: >-
+      Use `Enum.map_join/3` is more efficient than using `Enum.map/2 |> Enum.join/2`.
```
Comment (colleend) [inline elixir/lang/best-practice/enum.yaml]:
```suggestion
      Using `Enum.map_join/3` is more efficient than using `Enum.map/2 |> Enum.join/2`.
```

**Review by colleend — CHANGES_REQUESTED** [review]
Thank you for submitting this rule! We really appreciate all contributions. I've left a few comments for mostly formatting issues -- would be great if you could change the PR accordingly!

**Comment by wingyplus** [comment]
@colleend done. :)

**Review by colleend — APPROVED** [review]
lgtm! thank you again!

**Pull request merged**